docs/reference/runtime/runtime-class.md

```diff
@@ -77,7 +77,77 @@ runtime.registerRoutes([
+[!ref text="Setup the host application to accept hoisted routes"](/reference/routing/useHoistedRoutes.md)
+
+### Register routes under a specific nested layout route
+
+React router [nested routes](https://reactrouter.com/en/main/start/tutorial#nested-routes) enable applications to render nested layouts at various points within the router tree. This is quite helpful for federated applications as it enables composable and decoupled UI.
+
+To fully harness the power of nested routes, the `registerRoutes` function allows a route to be registered **under any** previously registered **nested layout route**, even if that route was registered by another module.
+
+When registering a new route with the `registerRoutes` function, to render the route under a specific nested layout route, specify a `layoutPath` property that matches the nested layout route's `path` property. The only requirement is that the **nested layout route** must has been **registered** to `@squide` **before** the **new child route**.
+
+```tsx !#10
+import { lazy } from "react";
+
+const Page = lazy(() => import("./Page.tsx"));
+
+runtime.registerRoutes([
+    {
+        path: "/layout/page-1",
+        element: <Page />
+    }
+], { layoutPath: "/layout" }); // Register the page under the "/layout" nested layout.
+```
+
+!!!info
+Likewise any other React Router routes, the `path` property of a page rendered under a nested layout must be an absolute path. For example, if a nested layout `path` is `/layout`, the `path` property of a page rendered under that layout route and responding to the `/page-1` url, should be `/layout/page-1`.
+!!!
+
+#### Index routes
+
+Although nested layout routes that serves as indexes (e.g. `{ index: true, element: <Layout /> }`) are not very common, `@squide` still supports this scenario. To register a route **under an index route**, set the `layoutPath` property as the concatenation of the index route's parent path and `/$index$`. 
+
+```tsx !#8,12 host/src/register.tsx
+import { lazy } from "react";
+
+const Page = lazy(() => import("./Page.tsx"));
+const Layout = lazy(() => import("./Layout.tsx"));
+
+runtime.registerRoutes([
+    {
+        path: "/page-1",
+        element: <Page />,
+        children: [
+            {
+                index: true,
+                element: <Layout />
+            }
+        ]
+    }
+]);
+```
+
+```tsx !#10 remote-module/src/register.tsx
+import { lazy } from "react";
+
+const Page = lazy(() => import("./Page.tsx"));
+
+runtime.registerRoutes([
+    {
+        path: "/page-1/page-2",
+        element: <Page />
+    }
+], { layoutPath: "/page-1/$index$" }); // Using $index$ to match "index: true"
+```
+
+### Retrieve routes
+
+A federated application routes are accessible from a `Runtime` instance, but keep in mind that the preferred way to retrieve the routes is with the [useRoutes](./useRoutes) hook.
+
+```tsx
+const routes = runtime.routes;
```

Comment:
Oof, we should highly discourage accessing routes directly like that. 

Related question, is there a useRuntime hook?

Reply:
Yes: https://gsoft-inc.github.io/wl-squide/reference/runtime/useruntime/

It's indeed not the preferred way to do it but it can be useful in context when a hook is not usable, generally in instrumentation code.